app/models/endpoint.rb

```diff
@@ -9,6 +9,19 @@ class Endpoint < ApplicationRecord
   validates :url, :uniqueness => true, :if => :url
   validate :validate_certificate_authority
 
+  after_create  :endpoint_created
+  after_destroy :endpoint_destroyed
+
+  def endpoint_created
+    # Make sure monitoring manager is created for the prometheus endpoint
```

Comment:
Since this is a generic callback that could be used for other things I don't think you need to call out the monitoring manager or prometheus endpoint specifically here and [here](https://github.com/ManageIQ/manageiq/pull/16635/files#diff-5d3f1a5ab701ca4a756d3c921e36e9f6R21)

Reply:
Right, removing.